lib/src/http/managers/message_manager.dart

```diff
@@ -166,9 +167,14 @@ class MessageManager extends Manager<Message> {
   }
 
   Reaction parseReaction(Map<String, Object?> raw) {
+    final bool isUnicode = (raw['emoji'] as Map<String, Object?>)['id'] == null;
+
+    final parseEmoji = isUnicode ? client.emojis.parseText : client.emojis.parse;
+
```

Comment:
The logic of choosing whether to parse as a text or guild emoji should be moved to the `parse` method of `EmojiManager` - see `ChannelManager.parse` for an example.